rudder-core/src/main/scala/com/normation/rudder/db/DB.scala

```diff
@@ -0,0 +1,365 @@
+import com.normation.rudder.domain.policies.Rule
+import com.normation.rudder.domain.policies.RuleId
+import com.normation.rudder.domain.policies.RuleTarget
+import com.normation.rudder.domain.reports.NodeConfigId
+import com.normation.rudder.domain.reports.Reports
+import com.normation.rudder.reports.execution.{ AgentRun => RudderAgentRun }
+import com.normation.rudder.reports.execution.AgentRunId
+import com.normation.rudder.repository.GitCommitId
+import com.normation.rudder.rule.category.RuleCategoryId
+
+import org.joda.time.DateTime
+
+
+
+
+
+/*
+ * Here, we need to declare all database table as a case classes that
+ * is understandable by doobie
+ *
+ * Actually, we don't need to declare ALL table, only part that are used
+ * for requests and are better named than let as TupleNN(...).
+ *
+ * Convention: all case classes declared here are under the DB object
+ * and should always be used with that prefix, for ex:
+ *
+ * ExpectedReports
+ *
+ * (i.e, we should never do "import _"
+ */
+
+final object DB {
+
+  //////////
+
+  final case class MigrationEventLog[T](
+      id                 : T
+    , detectionTime      : DateTime
+    , detectedFileFormat : Long
+    , migrationStartTime : Option[DateTime]
+    , migrationEndTime   : Option[DateTime]
+    , migrationFileFormat: Option[Long]
+    , description        : Option[String]
+  )
+
+  //////////
+
+  final case class ExpectedReports[T](
+      pkId                      : T
+    , nodeJoinKey               : Int
+    , ruleId                    : String
+    , serial                    : Int
+    , directiveId               : String
+    , component                 : String
+    , cardinality               : Int
+    , componentsValues          : String
+    , unexpandedComponentsValues: String
+    , beginDate                 : DateTime
+    , endDate                   : Option[DateTime]
+  )
+
+  //////////
+
+  final case class ExpectedReportsNodes(
+      nodeJoinKey        : Int
+    , nodeId             : String
+    , nodeConfigVersions : List[String]
+  )
+
+  //////////
+
+  final case class Reports[T](
+      id                 : T
+    , executionDate      : DateTime
+    , nodeId             : String
+    , directiveId        : String
+    , ruleId             : String
+    , serial             : Int
+    , component          : String
+    , keyValue           : String
+    , executionTimestamp : DateTime
+    , eventType          : String
+    , policy             : String
+    , msg                : String
+  )
+
+  //////////
+
+  final case class GitCommitJoin (
+      gitCommit     : GitCommitId
+    , modificationId: ModificationId
+  )
+
+  //////////
+
+  final case class RunProperties (
+      name : String
+    , value: String
+  )
+
+  //////////
+
+  final case class AgentRun(
+      nodeId      : String
+    , date        : DateTime
+    , nodeConfigId: Option[String]
+    , isCompleted : Boolean
+    , insertionId : Long // PURPOSE ?
```

Comment:
insertionId contains the id of the first received report with this date. The goal is to know which is the last reports we received from node (as network is most of the time more reliable than date ...)